server/controllers/oidc.js

```diff
@@ -18,9 +19,26 @@ const oidcSignIn = async (ctx) => {
   const { state } = ctx.query;
   const { OIDC_CLIENT_ID, OIDC_REDIRECT_URI, OIDC_SCOPES, OIDC_AUTHORIZATION_ENDPOINT } = configValidation();
 
-  const authorizationUrl = `${OIDC_AUTHORIZATION_ENDPOINT}?response_type=code&client_id=${OIDC_CLIENT_ID}&redirect_uri=${OIDC_REDIRECT_URI}&scope=${OIDC_SCOPES}&state=${state}`;
+  // Generate code verifier and code challenge
+  const { code_verifier: codeVerifier, code_challenge: codeChallenge } =
+    pkceChallenge();
 
-  ctx.redirect(authorizationUrl);
+  // Store the code verifier in the session
+  ctx.session.codeVerifier = codeVerifier;
+
+  const params = new URLSearchParams();
+  params.append('response_type', 'code');
+  params.append('client_id', OIDC_CLIENT_ID);
+  params.append('redirect_uri', OIDC_REDIRECT_URI);
+  params.append('scope', OIDC_SCOPES);
+  params.append('code_challenge', codeChallenge);
+  params.append('code_challenge_method', 'S256');
+  if (state) {
+    params.append('state', state);
+  }
+  const authorizationUrl = `${OIDC_AUTHORIZATION_ENDPOINT}?${params.toString()}`;
+  ctx.set('Location', authorizationUrl);
+  return ctx.send({}, 302);
```

Comment:
`ctx.redirect` generates a useless response body, so I fixed it to match the implementation of other providers.